src/torchmetrics/functional/audio/dnsmos.py

```diff
@@ -0,0 +1,268 @@
+    `DNSMOS P835 <https://arxiv.org/abs/2110.01763>`_.
+
+
+    .. note:: using this metric requires you to have ``librosa``, ``onnxruntime`` and ``requests`` installed.
+        Install as ``pip install torchmetrics['audio']`` or alternatively `pip install librosa onnxruntime-gpu requests`
+        (if you do not have GPU enabled machine install `onnxruntime` instead of `onnxruntime-gpu`)
+
+    Args:
+        preds: [..., time]
+        fs: sampling frequency
+        personalized: whether interfering speaker is penalized
+        device: the device used for calculating DNSMOS, can be cpu or cuda:n, where n is the index of gpu.
+            If None is given, then the device of input is used.
+        num_threads: the number of threads to use for cpu inference. Defaults to None.
+
+    Returns:
+        Float tensor with shape ``(..., 4)`` of DNSMOS values per sample, i.e. [p808_mos, mos_sig, mos_bak, mos_ovr]
+
+    Raises:
+        ModuleNotFoundError:
+            If ``librosa``, ``onnxruntime`` or ``requests`` packages are not installed
+
+    Example:
+        >>> from torch import randn
+        >>> from torchmetrics.functional.audio.dnsmos import deep_noise_suppression_mean_opinion_score
+        >>> g = torch.manual_seed(1)
+        >>> preds = randn(8000)
+        >>> deep_noise_suppression_mean_opinion_score(preds, 8000, False)
+        tensor([2.2285, 2.1132, 1.3972, 1.3652], dtype=torch.float64)
+
+    """
+    if not _LIBROSA_AVAILABLE or not _ONNXRUNTIME_AVAILABLE or not _REQUESTS_AVAILABLE:
+        raise ModuleNotFoundError(
+            "DNSMOS metric requires that librosa, onnxruntime and requests are installed."
+            " Install as `pip install librosa onnxruntime-gpu requests`."
+        )
+    device = torch.device(device) if device is not None else preds.device
+
+    onnx_sess = _load_session(f"{DNSMOS_DIR}/{'p' if personalized else ''}DNSMOS/sig_bak_ovr.onnx", device, num_threads)
+    p808_onnx_sess = _load_session(f"{DNSMOS_DIR}/DNSMOS/model_v8.onnx", device, num_threads)
+
+    desired_fs = SAMPLING_RATE
+    if fs != desired_fs:
+        audio = librosa.resample(preds.cpu().numpy(), orig_sr=fs, target_sr=desired_fs)
+    else:
+        audio = preds.cpu().numpy()
+
+    len_samples = int(INPUT_LENGTH * desired_fs)
+    while audio.shape[-1] < len_samples:
+        audio = np.concatenate([audio, audio], axis=-1)
+
+    num_hops = int(np.floor(audio.shape[-1] / desired_fs) - INPUT_LENGTH) + 1
+
+    moss = []
+    hop_len_samples = desired_fs
+    for idx in range(num_hops):
+        audio_seg = audio[..., int(idx * hop_len_samples) : int((idx + INPUT_LENGTH) * hop_len_samples)]
+        if audio_seg.shape[-1] < len_samples:
+            continue
+        shape = audio_seg.shape
+        audio_seg = audio_seg.reshape((-1, shape[-1]))
+
+        input_features = np.array(audio_seg).astype("float32")
+        p808_input_features = np.array(_audio_melspec(audio=audio_seg[..., :-160])).astype("float32")
```

Comment:
I believe `audio_seg` is always a Numpy array already.
Are these type conversions necessary?